### ISSUE TITLE
Provide built-in Popper for Tooltip

### ISSUE THREAD

**Issue by anicholls** [Workday/canvas-kit#86]
## 🚀 Feature Proposal

Create a new convenience component that wraps `Tooltip` with our [`Popper` helper component](https://github.com/Workday/canvas-kit/blob/master/modules/common/react/lib/Popper.tsx).

This change could be additive by simply overlaying the `Popper` props on top of `Tooltip`. If the `Popper` props aren't used, then the Popper.js positioning engine wouldn't be used and the consumer could implement their own positioning engine if desired.

In addition, the functionality for triggering and hiding the tooltip on hover/focus/blur should be implemented.

## Motivation

To provide the consumer with a more robust and easier to consume tooltip experience. 

## Example

```
<Tooltip placement={'right'} anchorElement={this.buttonRef.current}>
  Close
</Tooltip>
```

**Comment by davvidbaker** [comment]
Would/could this bring an optional arrow to the tooltip? I'm looking at a design like 👇  and I don't think the current tooltip supports that.
![image](https://user-images.githubusercontent.com/6373939/64999210-23e5a700-d8a4-11e9-809c-73036750cb49.png)

**Comment by anicholls** [comment]
Hi @davvidbaker

This was actually removed from our design system earlier this year. The decision was made that the cursor position and animation origin were enough to indicate the source of the tooltip. I would let your designers know that the tooltip arrows are no longer supported in the Canvas spec, and they should just be using the plain rectangle.

Let me know if you have any questions!

**Comment by mxkellman** [comment]
Heya @anicholls - I think we do need the arrow in some cases.  When you have a toolbar for example, it's very helpful to clearly point to the icon that is currently being hovered.  The toolbar here in Github has it, for example.  In the example above @davvidbaker is only showing the first phase of the feature with just one icon.  Here's what it looks like in the more developed designs.

<img width="1257" alt="Screen Shot 2019-09-17 at 1 43 12 PM" src="https://user-images.githubusercontent.com/3118515/65074041-4255ac00-d951-11e9-8a0f-e39034a84ccd.png">

The toolbar button styling be used for these buttons and ideally toolbar buttons would have tooltips simply built in since we should have tooltips by default for those since some of them can be ambiguous.

**Comment by anicholls** [comment]
Hey @mxkellman!

Use cases like these were definitely considered when making the call to drop the arrow. The consensus was that the tooltip is just telling you what will happen if you click where your mouse is (or hit space on what you're focussed on). Your cursor/focus outline naturally acts as a way to connect the tooltip with the element and it is redundant to point at the thing your mouse is already hovering over or that you're already focused on. Even when there are adjacent buttons/actionable elements, this still remains the case. @jessespencer feel free to chime in here if you'd like.

Most teams have been alright with this direction, but based on your concerns we will definitely get this in the pipeline for user testing. Until we have conclusive results against this from testing we're going to stick with no arrows in Canvas Kit. Thanks for your thoughts though!

**Comment by mkellman-workday** [comment]
@anicholls Thanks for the thoughtful reply.  I switched over to my Workday account - sorry for any confusion.  A few more things for your consideration.  The button outline on the toolbar item hover does not provide a lot of contrast (Soap 300), most icons don't take up the whole space, and the buttons are very close together. The net result is that it's easy to think you're hovered over one item when you're actually hovered over another.  The position of the tooltip helps, but for longer names "Background Fill" there is still some ambiguity.  Thanks for putting this in the pipeline for testing!

**Comment by NicholasBoll** [comment]
Closed by #528
